src/main/java/org/mvel2/execution/ExecutionArrayList.java

```diff
@@ -116,13 +165,134 @@ public long memorySize() {
         return memorySize;
     }
 
+    public int indexOf(Object o, int fromIndex) {
+        int idx = this.slice(fromIndex).indexOf(o);
+        return idx == -1 ? idx : idx + fromIndex;
+    }
+
     public String join() {
         return join(",");
     }
 
     public String join(String separator) {
-        return this.stream()
-                .map( Object::toString )
-                .collect( Collectors.joining( separator ) );
+        List rez = new ArrayList<String>();
```

Comment:
Please leave the previous implementation of the method.

Reply:
This will result in a non-JS implementation^